packages/material-ui/src/ButtonBase/ButtonBase.js

```diff
@@ -53,43 +47,71 @@ export const styles = {
   focusVisible: {},
 };
 
+/**
+ * https://github.com/facebook/react/issues/14099#issuecomment-440013892
+ *
+ * @param {function} fn
+ */
+function useEventCallback(fn) {
+  const ref = React.useRef(fn);
+  const useEnhancedEffect = typeof window !== 'undefined' ? React.useLayoutEffect : React.useEffect;
+  useEnhancedEffect(() => {
+    ref.current = fn;
+  });
+  return React.useCallback(event => (0, ref.current)(event), []);
```

Comment:
```suggestion
  return React.useCallback(event => ref.current(event), []);
```

Reply:
https://stackoverflow.com/a/32275207/3406963

Reply:
I don't have the full picture but I would trust Sophie to not post wrong code. Better stick with the source code instead of changing things we do not fully understand.